money-core/src/main/scala/com/comcast/money/core/formatters/IngressFormatter.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2012 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.core.formatters
+import com.comcast.money.api.SpanId
+
+/**
+ * Decorates a formatter so that it interprets headers for incoming requests but does not write headers for outgoing requests.
+ */
+final class IngressFormatter(val formatter: Formatter) extends Formatter {
+  override def toHttpHeaders(spanId: SpanId, addHeader: (String, String) => Unit): Unit = ()
```

Comment:
I think I would rather fail loudly if someone calls this than do nothing silently

Reply:
They're supposed to be called, in that Formatters assume that they both read and write headers.  But for these two Formatters they're special in that they either read or write headers, not both.   But the Formatters they wrap don't know this.

Reply:
I get that, I guess my concern would be a user calls the wrong method and gets not feedback telling them that it's wrong

Reply:
A user isn't going to call them directly, instrumentation is and it won't know that the formatter isn't supposed to read or write certain headers, especially if these formatters are used as a part of a FormatterChain.

Reply:
gotcha!

Reply:
can we make them private to money?